### PR TITLE
[MA-405] SDK update

### PR DESCRIPTION
**Pull request by sukrit-ym**
Update the min SDK version to 21, (Promise in java script is not supported below it)
Remove redundant permissions and code
Put runtime permission in place
Update SDK to work on Android 10+
Make resources private
Update documentation with dependency and permissions used in app

### CODE REVIEW

File: YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java [hunk 1]
```diff
@@ -192,6 +213,19 @@ public void onCreate(Bundle savedInstanceState) {
                         }
                     }
                     break;
+                case "image-opened":
```
Comment (kaunteya) [inline YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java:216]: As discussed, please make sure events are going to the host app

**Review by kaunteya — APPROVED** [review]

File: YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java [hunk 2]
```diff
@@ -93,10 +93,10 @@ public void closeBot() {
 
     public void emitEvent(YMBotEventResponse event) {
         if (event != null) {
-            if (botCloseEventListener != null && event.getCode() != null && event.getCode().equals("bot-closed")) {
+            if (botCloseEventListener != null && event.getCode() != null && isCloseBotEvent(event)) {
                 botCloseEventListener.onClosed();
             } else {
-                if (listener != null)
+                if (listener != null && !(isImageOpenEvent(event) || isImageCloseEvent(event)))
```
Comment (sukrit-ym) [inline YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:99]: @kaunteya  this is the check that will make sure, we don't pass "image-opened" and "image-closed" events to client.

Comment (PriyankUpadhyay) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:99]: Hey @sukrit-ym can we have something generic to identify if the event is to be percolated to the host app or not? We can use a special parameter in the data part of the event, based on which we can filter such an event.

Comment (kaunteya) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java:99]: That's a good point priyank. 
The entire info comes in "ymHandler" probably we can have another key at a top level saying "ymInternalEvent"

Comment (sukrit-ym) [reply YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java]: done

**Pull request merged**